UIView+MGBadgeView.m

```diff
@@ -83,9 +85,7 @@ - (void)setBadgeValue:(NSInteger)badgeValue {
         
         if(badgeValue != 0 || _displayIfZero) {
             [self mg_updateBadgeViewSize];
-        
-            if(_position == MGBadgePositionBest)
-                [self mg_updateBadgeViewPosition];
+            [self mg_updateBadgeViewPosition];
```

Comment:
This is not needed if the position isn't best.
You probably want put back the old code and add a call to `mg_updateBadgeViewSize` (for all the changes from here to the end)